modules/ipc/src/zenoh/liveliness.cpp

```diff
@@ -193,16 +193,23 @@ void EndpointDiscovery::createLivelinessSubscriber() {
   auto liveliness_callback = [this](const ::zenoh::Sample& sample) mutable {
     auto info = parseLivelinessToken(sample.get_keyexpr().as_string_view(), sample.get_kind());
     if (info.has_value() && topic_filter_.isAcceptable(info->topic)) {
-      infos_consumer_->queue().forcePush(std::move(*info));
+      auto dropped_element = infos_consumer_->queue().forcePush(std::move(*info));
+      if (dropped_element.has_value()) {
+        heph::log(heph::ERROR,
+                  "Dropped endpoint discovery info before being able to process due to full queue",
+                  "dropped_info", fmt::to_string(dropped_element.value()));
```

Comment:
in order for this to work you need to include `hephaestus/format/generic_formatter.h` and you do not need `fmt::to_string`